plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -823,6 +852,87 @@ public Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session,
         return Optional.of(new ConnectorTableLayout(partitionColumnNames));
     }
 
+    @Override
+    public Iterator<RelationCommentMetadata> streamRelationComments(ConnectorSession session, Optional<String> schemaName, UnaryOperator<Set<SchemaTableName>> relationFilter)
+    {
+        List<RelationCommentMetadata> views = getViews(session, schemaName).entrySet().stream()
+                .map(entry -> RelationCommentMetadata.forRelation(entry.getKey(), entry.getValue().getComment()))
+                .collect(toImmutableList());
```

Comment:
we are iterating the same structure twice. can we just use for loop instead?

something like:

```
        Map<SchemaTableName, ConnectorViewDefinition> viewDefinitions = getViews(session, schemaName);
        ImmutableList.Builder<RelationCommentMetadata> commentMetadataBuilder = ImmutableList.builderWithExpectedSize(viewDefinitions.size());
        ImmutableSet.Builder<SchemaTableName> viewNamesBuilder = ImmutableSet.builderWithExpectedSize(viewDefinitions.size());
        for (Entry<SchemaTableName, ConnectorViewDefinition> viewDefinitionEntry : viewDefinitions.entrySet()) {
            RelationCommentMetadata relationCommentMetadata = RelationCommentMetadata.forRelation(viewDefinitionEntry.getKey(), viewDefinitionEntry.getValue().getComment());
            commentMetadataBuilder.add(relationCommentMetadata);
            viewNamesBuilder.add(relationCommentMetadata.name());
        }
        List<RelationCommentMetadata> views = commentMetadataBuilder.build();
        Set<SchemaTableName> viewNames = viewNamesBuilder.build();
```